assembler.py

```diff
@@ -7,9 +7,109 @@
 parallel_mode = lambda x: x._parallel_mode
 
 
-def get_jobs(module):
+def get_tasks(module):
     tasks = sorted(
         [obj for obj in inspect.getmembers(module) if is_job(obj[1])],
         key=lambda x: parallel_mode(x[1])
     )
     return OrderedDict(tasks)
+
+
+class Assembly(object):
+    get_tasks = get_tasks
+
+    def __init__(self, module, *args, **kwargs):
+        self.module = module
+
+    def get_arguments_for_func(function_name):
+        """
+        Returns a tuple, dict pair.
+
+        This has to be implemented depending on what context this is running
+        in.
+
+        For example, the user may enter these on a form on the web UI.
+        """
+        pass
+
+    def assemble(self):
+        tasks = self.get_tasks(self.module)
+        for task in tasks:
+            arguments = self.get_arguments_for_func(task)
+            func = tasks[task]
+            tasks[task] = {'func': func, 'arguments': arguments}
```

Comment:
At the moment this will do, but I am not sure there is much relevance in `func.__name__` being a key. Perhaps we can just use func as the key and `__name__` where needed. The decorators should also be `wrapped` with `functools`.